python/dgl/data/tu.py

```diff
@@ -387,6 +387,14 @@ def _file_path(self, category):
     def _idx_from_zero(idx_tensor):
         return idx_tensor - np.min(idx_tensor)
 
+    @staticmethod
+    def _idx_reset(idx_tensor):
+        """Maps n unique labels to {0, ..., n-1} in an ordered fashion."""
+        labels = np.unique(idx_tensor)
+        relabel_map = {x: i for i, x in enumerate(labels)}
+        new_idx_tensor = np.vectorize(relabel_map.get)(idx_tensor)
```

Comment:
Just for curiosity: how did you experience the speed up by `np.vectorize`'ing the `get` method?  If the gain is substantial then we probably want to standardize it in the future.

Reply:
I didn't see how it compares to a list comprehension wrapped in `np.array` (I try and avoid type conversions if I can) but the docs for `np.vectorize` state its not designed for speed:

> The vectorize function is provided primarily for convenience, not for performance. The implementation is essentially a for loop.

There's some further discussion on stackoverflow which may be of interest:

+ https://stackoverflow.com/questions/35215161/most-efficient-way-to-map-function-over-numpy-array
+ https://stackoverflow.com/questions/16992713/translate-every-element-in-numpy-array-according-to-key